pyls/rpc_manager.py

```diff
@@ -82,7 +83,8 @@ def cancel(self, request_id):
         """
         log.debug('Cancel request %d', request_id)
         try:
-            self._received_requests[request_id].cancel()
+            self._received_requests[request_id].set_exception(
+                JSONRPCDispatchException(code=LSP_CANCEL_CODE, message="Request cancelled"))
```

Comment:
Hanging parenthesis!

Reply:
And can you put the request id in the message

Reply:
don't need the request id! the actual response is create here: https://github.com/palantir/python-language-server/blob/develop/pyls/rpc_manager.py#L150